sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubClientFactory.cs

```diff
@@ -48,14 +48,14 @@ internal EventHubProducerClient GetEventHubProducerClient(string eventHubName, s
             eventHubName = _nameResolver.ResolveWholeString(eventHubName);
             connection = _nameResolver.ResolveWholeString(connection);
 
-            return _producerCache.GetOrAdd(eventHubName, key =>
```

Comment:
Do we need the same logic for the consumerCache?

Reply:
Hmm, I did not dig into the `EventHubTriggerAttribute`, but I think a function can only have one eventhub trigger.  so the `_consumerCache` is bug free?

Reply:
I think the EventHubClientFactory is a singleton across an instance of a function app, so the same instance would be used for multiple functions. We should probably add a test to confirm this, but I suspect that if you have two functions using different namespaces, but same hub name you would hit the same issue.

Reply:
Fixed with the same logic.